recipes/openssh/files/init

```diff
@@ -32,18 +34,13 @@ check_config() {
 
 check_keys() {
 	# create keys if necessary
-	if [ ! -f /etc/ssh/ssh_host_rsa_key ]; then
-		echo "  generating ssh RSA key..."
-		ssh-keygen -q -f /etc/ssh/ssh_host_rsa_key -N '' -t rsa
-	fi
-	if [ ! -f /etc/ssh/ssh_host_dsa_key ]; then
-		echo "  generating ssh DSA key..."
-		ssh-keygen -q -f /etc/ssh/ssh_host_dsa_key -N '' -t dsa
-	fi
-	if [ ! -f /etc/ssh/ssh_host_ecdsa_key ]; then
-		echo "  generating ssh ECDSA key..."
-		ssh-keygen -q -f /etc/ssh/ssh_host_ecdsa_key -N '' -t ecdsa
-	fi
+	for keytype in $HOST_KEY_TYPES ; do
+		keyfile="/etc/ssh/ssh_host_"$keytype"_key"
+		if [ ! -f "$keyfile" ]; then
+			echo "  generating ssh $keytype key..."
+			ssh-keygen -q -f "$keyfile" -N '' -t rsa
```

Comment:
if you add a "sync" here, I can be lazy and not push an additional fixup :) Without, we can lose the key if we poweroff too early (empty file) and then it will not be generated